tests/OrganizationsMetadataTest.php

```diff
@@ -0,0 +1,175 @@
+        $this->client->removeUserFromOrganization($this->organization['id'], $this->superAdmin['id']);
+    }
+
+    public function testNormalUserCanNotManageMetadata(): void
+    {
+        $this->client->addUserToOrganization($this->organization['id'], ['email' => $this->superAdmin['email']]);
+        $organizationId = $this->organization['id'];
+
+        $this->createUserMetadata($this->client, $organizationId);
+
+        // Normal user should not ADD userMetadata
+        try {
+            $this->createUserMetadata($this->normalUserClient, $organizationId);
+            $this->fail('Test should not reach this line.');
+        } catch (ClientException $e) {
+            $this->assertEquals(403, $e->getCode());
+        }
+
+        // Normal user should not SEE org metadata
+        try {
+            $this->normalUserClient->listOrganizationMetadata($organizationId);
+            $this->fail('Test should not reach this line.');
+        } catch (ClientException $e) {
+            $this->assertEquals(403, $e->getCode());
+        }
+    }
+
+    public function testSuperAdminCanManageMetadata(): void
+    {
+        $this->client->addUserToOrganization($this->organization['id'], ['email' => $this->normalUser['email']]);
+        $organizationId = $this->organization['id'];
+
+        $metadata = $this->createUserMetadata($this->client, $organizationId);
+
+        $this->assertCount(2, $metadata);
+        $this->validateMetadataEquality(self::TEST_METADATA[1], $metadata[0], self::PROVIDER_USER);
+        $this->validateMetadataEquality(self::TEST_METADATA[0], $metadata[1], self::PROVIDER_USER);
+
+        $metadataArray = $this->client->listOrganizationMetadata($organizationId);
+
+        $this->assertCount(2, $metadataArray);
+        $this->validateMetadataEquality(self::TEST_METADATA[1], $metadataArray[0], self::PROVIDER_USER);
+        $this->validateMetadataEquality(self::TEST_METADATA[0], $metadataArray[1], self::PROVIDER_USER);
+    }
+
+    public function testMaintainerCanManageMetadata(): void
+    {
+        $this->client->addUserToMaintainer($this->testMaintainerId, ['email' => $this->normalUser['email']]);
+
+        $metadata = $this->createUserMetadata($this->normalUserClient, $this->organization['id']);
+        $this->assertCount(2, $metadata);
+        $this->validateMetadataEquality(self::TEST_METADATA[1], $metadata[0], self::PROVIDER_USER);
+        $this->validateMetadataEquality(self::TEST_METADATA[0], $metadata[1], self::PROVIDER_USER);
+
+        $metadataArray = $this->normalUserClient->listOrganizationMetadata($this->organization['id']);
+        $this->assertCount(2, $metadataArray);
+        $this->validateMetadataEquality(self::TEST_METADATA[1], $metadataArray[0], self::PROVIDER_USER);
+        $this->validateMetadataEquality(self::TEST_METADATA[0], $metadataArray[1], self::PROVIDER_USER);
+
+        $this->client->removeUserFromMaintainer($this->testMaintainerId, $this->normalUser['id']);
```

Comment:
Toto nemusis v tomto teste robit, pretoze v postate si zas spat spravis z maintainer admina normalneho usera a to uz testujes v provom teste ze to nejde

Reply:
radeji bych to explicitne pohlidal